### PR TITLE
Finished code for x shear!

### PR DESCRIPTION
**Pull request by AFraser16**

### CODE REVIEW

**Review by aidkvndfj — CHANGES_REQUESTED** [review]
Looks like there are some issues with compiling, I'm getting lots of warnings. Please fix and we'll go from there. If you need help getting the latest changes from master, let us know.

**Comment by aidkvndfj** [comment]
Please also change the folder name to reflect your role in changing the shape.

**Comment by AFraser16** [comment]
I think everything is fixed up now.

File: ShearInX/ShearInX.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#ifndef XSHEAR_H
+#define XSHEAR_H
+
+// Must always be defined, but only used in test mode
+int errorFlag;
```
Comment (cc0407) [inline ShearInX/ShearInX.h:5]: if theyre only used in test mode, you could probably move them to the ShearInXTests.h

**Review by cc0407 — CHANGES_REQUESTED** [review]
- Please remove the tstFlag if-else block in your tests.h file. 
- The test handler shouldn't run xShear if all of the tests are successful because we're running xShear later on in the program. 
- When I tried to remove the if-else block myself it segfaulted on your first clean test so im not sure if theres something bigger happening there.
- You shouldnt need to create test points because we will provide each group with their own clean slate of test points (we'll provide 5 points) so you can just remove your createTestPoints(); function.

Once you've implemented these just give it a quick once over with `make`, `./main`, `./main -t`, and the valgrind command from the readMe to make sure it runs without any warnings or errors!

**Comment by AFraser16** [comment]
Updated to further match requests

**Review by aidkvndfj — APPROVED** [review]

**Review by aidkvndfj — APPROVED** [review]
Everything looks good to me, and runs good.

**Review by cc0407 — APPROVED** [review]
Looks great! Runs without warning or error, and valgrind isnt complaining

**Pull request merged**